### ISSUE TITLE
Ability to modify expr_type of MediumLevelILInstruction

### ISSUE THREAD

**Issue by 0x1F9F1** [Vector35/binaryninja-api#1073]
Binary Ninja Version: 1.1.1236-dev, cb67afb0
Platform: Windows 10 (10.0)

Currently `MediumLevelILInstruction.expr_type` is read-only, but it would be great if we were able to modify it, such as when dealing with vtable calls since there is no variable you can modify to set the call type.
```
>>> current_mlil[7]
<il: rax_1 = [rax + 0x240].q(rcx, rdx, r8)>
>>> current_mlil[7].dest.expr_type
<type: int64_t, 0% confidence>
```

**Comment by rssor** [comment]
While updating the type of the IL expressions is not something we'll be able to support, there is a core-only API I'm looking into exposing called `Set{User/Auto}CallTypeAdjustment` which allows the type information of a call target to be overridden with an arbitrary function type, which I believe addresses the use-case.

**Comment by plafosse** [comment]
This will be fixed when we solve #1031
